testprojects/src/java/org/pantsbuild/example/ExampleLib.java

```diff
@@ -1,7 +1,29 @@
-package org.pantsbuild.example.lib;
+package org.pantsbuild.example;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
 
 public class ExampleLib {
-    public static String hello() {
-        return "Hello!";
+    private String template = "{\"contents\": \"Hello, World!\"}";
+
+    public String getGreeting() {
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            SerializedThing thing = mapper.readValue(template, SerializedThing.class);
+            return thing.contents;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void main(String[] args) {
+        System.out.println(new ExampleLib().getGreeting());
     }
```

Comment:
So, while it's always good to have some code for sanity checking, I think the thrust of the discussion in https://github.com/pantsbuild/pants/pull/12859#discussion_r707865241 was that we should not attempt to convert this into full blown test cases.

It's grey area, but IMO, adding a `fat_jar` target and `main` method makes sense because it gives you something to actually smoke test running `./pants package` on. But adding JSON output begins to cross over into what should be in a unit test instead.

Reply:
@stuhood The want from adding a `fat_jar` target here was to check that we correctly capture (transitive) dependencies when we do that, and trying to instantiate an object from a transitive dependency seemed like a good way to verify that. Do you have an alternative suggestion for what to put into a meaningful fat JAR?